torch/_inductor/ir.py

```diff
@@ -3028,6 +3028,41 @@ class InputBuffer(Buffer):
     pass
 
 
+@dataclasses.dataclass
+class WorkspaceBuffer(Buffer):
```

Comment:
In #117992 I also add workspace allocations but take a slightly different approach. The workspace size is an implementation detail of the code generation, so I don't represent it in the inductor IR and instead put it in `KernelArgs` and have the `Kernel` method create the args.

wdyt?

Reply:
@peterbell10 - Just looked at it, looks good to me. If the PR you mention is blocked through this ROCm issue (Is that blocking?),maybe you could split the workspace allocation code off into a separate PR, so we can get it merged quickly and I can adapt this to it.

Reply:
ROCm issues are fixed and I'm just waiting on perf results before merging.